### ISSUE TITLE
Support for Specifying Radarr/Sonarr Instance for Downloads in Multi-Instance Jellyseerr Setup

### ISSUE THREAD

**Issue by Chesyre** [psyko-gh/overcrawlrr#34]
Hi,

I am using Overcrawlrr to automate movie requests via rules and it has been very helpful. However, I have a multi-instance setup for Jellyseerr and I would like to know if it is possible to specify which Radarr or Sonarr instance should handle the download of a particular movie.

Currently, Overcrawlrr does not seem to have an option to direct requests to specific instances, which would be very useful in a multi-instance environment to better manage and distribute downloads.

Requested Feature:

An option in the rule configuration to specify the Radarr/Sonarr instance that should handle the request.
Integration with multi-instance Jellyseerr setups to ensure requests are sent to the correct instance.
Use Case: In a multi-instance setup, different instances may be configured for different genres, qualities, or sources. Being able to specify the instance would allow for more granular control over where movies are downloaded and how they are managed.

Expected Outcome:

A new field in the rule configuration UI where the user can select the specific Radarr/Sonarr instance.
Enhanced integration with Jellyseerr to support multi-instance configurations.
Thank you for considering this feature request!

Best regards,

**Comment by psyko-gh** [comment]
Hi,

You're right, Overcrawlrr only support a single Overseerr/Jellyseerr setup for now.
I'm quite confuse: do you have multiple Jellyseer instance, or multiple Radarr instances (or both) ?

So there are 2 requests:
1. feature 1: allow the configuration of multiple Jellyseerr servers,
2. feature 2: allow to specify which instance of Radarr to use in a specific Jellyseerr server

Is that correct ?

For the time being, Overcrawlrr does not support TV shows, so Sonarr is not yet relevant in this issue.

**Comment by destrodxbad** [comment]
I apologize for the confusion in my initial request. Let me clarify:
To be precise:

1. I have only one instance of Jellyseerr.
2. Within this single Jellyseerr instance, I have three different Radarr instances configured.

My request is specifically about the second feature you mentioned:

Feature 2: Allow specifying which Radarr instance to use within the single Jellyseerr server.

The reason for this setup is that I use different Radarr instances for different types of content. For example:

-One Radarr instance for documentaries
-Another Radarr instance for feature films

It would be very useful if Overcrawlrr could allow us to define a specific Radarr server for downloads based on rules. 
This way, we could automate requests to go to the appropriate Radarr instance based on the type of content.

**Comment by psyko-gh** [comment]
No worries, thx for the clarification.
Your request makes sense. I'm just wondering at which level should this be specified ?

- **at ruleset level**: the target Radarr to use would be configuring for an entire ruleset. Whenever a rule action is `accept`, the request would be made using the parent ruleset parameter. But this could be kind of problematic when a ruleset extends another.
- **at rule level**: each rule with an `accept` action could configure the target Radarr server. This would be the most flexible option, but also the most verbose.

Could you please share your rules configuration to show me on which basis you would made the decision of the Radarr server to chose ?

At the rule level, the setting could be something like:
```yml
  rules:
    - name: Acceptable sci-fi movie
      whenMatch:
        - genre:
            - science-fiction
        - score: above 5
      action: accept
      with:
        - radarr: the-name-of-the-instance
```

This feature might also be the opportunity to add a `4k` settings, since Overseerr (and probably Jellyseerr) supports 2 default Radarr server: one for 4K and one for non-4K

**Comment by destrodxbad** [comment]
I agree that implementing it at the rule level would be the most flexible way to go.

Here's my current config:

```yaml
config:
  overseerr:
    apiUrl: http://jellyseerr:5055/api/v1
    user: '****'
    password: '****'
    dryRun: false
  discovery:
    cron: '0 3 * * *'
    streams:
      - upcoming
      - popular
      - trending
    ruleset: films
  rulesets:
    - name: films
      extends: documentaires
      rules:
        - name: Accepter tous les films
          whenMatch:
            - adult: no
          action: accept
    - name: documentaires
      extends: spectacles
      rules:
        - name: Refuser les documentaires
          whenMatch:
            - genre:
                - Documentaire
          action: reject
    - name: spectacles
      rules:
        - name: Refuser les spectacles
          whenMatch:
            - keyword:
              - one-man show
              - one woman show
              - sketch
          action: reject
        - name: Refuser les films sans writer
          whenMatch:
              - crew:
                  job: Writer
                  names:
                    - "*"
          action: reject
```
          
Right now, as you can see, I've got a rule that accepts all movies and filters out documentaries and shows to download only movies.

**Comment by psyko-gh** [comment]
I will probably try and implement this radarr selection at rules level.
In your example, it means that the `films` ruleset won't be able to extend `documentaires` anymore, as you probably will have to configure `documentaires` to use the `radarr-documentaires` _(or whatever you named it)_

As for the support of Sonarr, I imagine it's possible. I simply did not implement it because I had no need for it. 
I will give it a try, but I can't give you any date on this...

**Comment by destrodxbad** [comment]
Ok, great. 
Thanks for your work on this!

**Comment by psyko-gh** [comment]
The feature is available on the edge image: https://github.com/psyko-gh/overcrawlrr/pkgs/container/overcrawlrr/281153428?tag=edge

The documentation is available here: https://psyko-gh.github.io/overcrawlrr/edge/rulesets/#rule-request-options

I'll let you beta test the feature. When you confirm it works as expected, I'll release it

**Comment by destrodxbad** [comment]
I have tested the new version and did not encounter any errors. The movies are correctly sent to the right Radarr server, and my Radarr servers are properly detected in the logs.

However, there is an issue with the processing of movie results. Only the first 45 results from the "upcoming," "popular," and "trending" categories are being processed, and these results do not include documentaries or shows. This limitation prevents me from testing the sending of movies to other servers.

For example, when I access the following URL:

```yaml
http://localhost:5056/api/movies/<id_movie>/evaluate/<ruleset_name>
```

it only performs a dry run and does not allow me to test a specific documentary to see if the rule works.

Given this, I wonder if it is possible to configure the number of movies retrieved. Specifically, can we modify the current limit of 45 results to retrieve, for instance, the first 100 results instead?

```log
[ 2024-09-30 12:17:03.434 ] info: ✔  Overseerr connection successful

[ 2024-09-30 12:21:51.817 ] info: Starting discovery...

[ 2024-09-30 12:21:52.234 ] info: Using ruleset: films

[ 2024-09-30 12:21:52.234 ] info:  Searching into following streams: upcoming, popular, trending

[ 2024-09-30 12:21:52.299 ] info: Found 45 to analyze...

[ 2024-09-30 12:21:52.300 ] info:   Skipping  - "S.O.S. Bikini Bottom : Une mission pour Sandy Écureuil" because it has already been processed

[ 2024-09-30 12:21:52.300 ] info:   Skipping  - "Joker: Folie à Deux" because it has already been processed

[ 2024-09-30 12:21:52.300 ] info:   Skipping  - "Venom: The Last Dance" because it has already been processed

[ 2024-09-30 12:21:52.301 ] info:   Skipping  - "Sonic 3, le film" because it has already been processed

[ 2024-09-30 12:21:52.301 ] info:   Skipping  - "Vaiana 2" because it has already been processed

[ 2024-09-30 12:21:52.301 ] info:   Skipping  - "Nom de code : Rouge" because it has already been processed

[ 2024-09-30 12:21:52.301 ] info:   Skipping  - "Terrifier 3" because it has already been processed

[ 2024-09-30 12:21:52.301 ] info:   Skipping  - "À contre-sens 2" because it has already been processed

[ 2024-09-30 12:21:52.302 ] info:   Skipping  - "John Wick : Ballerina" because it has already been processed

[ 2024-09-30 12:21:52.302 ] info:   Skipping  - "Miraculous World : London" because it has already been processed

[ 2024-09-30 12:21:52.302 ] info:   Skipping  - "Mufasa : Le Roi Lion" because it has already been processed

[ 2024-09-30 12:21:52.302 ] info:   Skipping  - "Gladiator II" because it has already been processed

[ 2024-09-30 12:21:52.302 ] info:   Skipping  - "Kraven the Hunter" because it has already been processed

[ 2024-09-30 12:21:52.302 ] info:   Skipping  - "Turno nocturno" because it has already been processed

[ 2024-09-30 12:21:52.302 ] info:   Skipping  - "Le Seigneur des Anneaux : La Guerre des Rohirrim" because it has already been processed

[ 2024-09-30 12:21:52.303 ] info:   Skipping  - "Smile 2" because it has already been processed

[ 2024-09-30 12:21:52.303 ] info:   Skipping  - "Mission Impossible : Dead Reckoning, partie 2" because it has already been processed

[ 2024-09-30 12:21:52.303 ] info:   Skipping  - "Dagr" because it has already been processed

[ 2024-09-30 12:21:52.303 ] info:   Skipping  - "Salsa Ni L" because it has already been processed

[ 2024-09-30 12:21:52.303 ] info:   Skipping  - "Criminal Squad : Pantera" because it has already been processed

[ 2024-09-30 12:21:52.303 ] info:   Skipping  - "Deadpool & Wolverine" because it has already been processed

[ 2024-09-30 12:21:52.303 ] info:   Skipping  - "The Crow" because it has already been processed

[ 2024-09-30 12:21:52.304 ] info:   Skipping  - "Moi, Moche et Méchant 4" because it has already been processed

[ 2024-09-30 12:21:52.304 ] info:   Skipping  - "Vice-versa 2" because it has already been processed

[ 2024-09-30 12:21:52.304 ] info:   Skipping  - "Jamais plus - It Ends with Us" because it has already been processed

[ 2024-09-30 12:21:52.305 ] info:   Skipping  - "The Substance" because it has already been processed

[ 2024-09-30 12:21:52.305 ] info:   Skipping  - "Hounds of War" because it has already been processed

[ 2024-09-30 12:21:52.305 ] info:   Skipping  - "Blink Twice" because it has already been processed

[ 2024-09-30 12:21:52.305 ] info:   Skipping  - "Le Robot sauvage" because it has already been processed

[ 2024-09-30 12:21:52.305 ] info:   Skipping  - "Hijack 1971" because it has already been processed

[ 2024-09-30 12:21:52.305 ] info:   Skipping  - "Beetlejuice Beetlejuice" because it has already been processed

[ 2024-09-30 12:21:52.305 ] info:   Skipping  - "Borderlands" because it has already been processed

[ 2024-09-30 12:21:52.305 ] info:   Skipping  - "Bad Boys: Ride or Die" because it has already been processed

[ 2024-09-30 12:21:52.306 ] info:   Skipping  - "L’I.A. du Mal" because it has already been processed

[ 2024-09-30 12:21:52.306 ] info:   Skipping  - "Profession ceinture noire" because it has already been processed

[ 2024-09-30 12:21:52.306 ] info:   Skipping  - "Subservience" because it has already been processed

[ 2024-09-30 12:21:52.306 ] info:   Skipping  - "Transformers : Le Commencement" because it has already been processed

[ 2024-09-30 12:21:52.306 ] info:   Skipping  - "Bandida - A Número Um" because it has already been processed

[ 2024-09-30 12:21:52.307 ] info:   Skipping  - "Sans un Bruit : Jour 1" because it has already been processed

[ 2024-09-30 12:21:52.307 ] info:   Skipping  - "Wolfs" because it has already been processed

[ 2024-09-30 12:21:52.307 ] info:   Skipping  - "Killer Heat" because it has already been processed

[ 2024-09-30 12:21:52.307 ] info:   Skipping  - "L' Appartement 7A" because it has already been processed

[ 2024-09-30 12:21:52.307 ] info:   Skipping  - "Rez Ball" because it has already been processed

[ 2024-09-30 12:21:52.333 ] info: ✔  Adding - "抓娃娃" because it matches rule "Accepter tous les films"

[ 2024-09-30 12:21:52.334 ] info: Fetching list of Radarr services...

[ 2024-09-30 12:21:53.476 ] info: Found Radarr server Film with profiles 1080p

[ 2024-09-30 12:21:53.476 ] info: Found Radarr server Documentaire with profiles 1080p

[ 2024-09-30 12:21:53.476 ] info: Found Radarr server Spectacle with profiles 1080p

[ 2024-09-30 12:21:53.540 ] info:   Skipping  - "Megalopolis" because it has already been processed

[ 2024-09-30 12:21:53.540 ] info: Discovery complete !
```

**Comment by psyko-gh** [comment]
So shows will not be evaluated until I had the support for TV shows.

The `/api/movies/<id_movie>/evaluate/<ruleset_name>` will indeed only evaluate the movie against the ruleset, without triggering an actual request to Overseerr. It is mainly used as a test tool.

For the limited results, there is currently no way to configure the number of movies to evaluate at each run. It runs daily on my side, therefore the 50ish movie limits is more than enough. The discovery job is mainly designed to monitor new movies. I'll see if I can add such a configuration, but I want to keep the configuration simple.

**Comment by destrodxbad** [comment]
> Shows will not be evaluated until I had the support for TV shows.

Oops, wrong translation, I meant to say "performance"

> For the limited results, there is currently no way to configure the number of movies to evaluate at each run. It runs daily on my side, therefore the 50ish movie limits is more than enough. The discovery job is mainly designed to monitor new movies. I'll see if I can add such a configuration, but I want to keep the configuration simple.

 I understand
Thanks anyway for this addition, I'll come back to you if I run into any problems.

**Comment by psyko-gh** [comment]
Available in 1.2.0
